### ISSUE TITLE
Is there any way to disable archiving?

### ISSUE THREAD

**Issue by mzwettler2** [CrunchyData/postgres-operator#3432]
I found PGO making archiving mandatory:
```
        // Send WAL files to all configured repositories when not in recovery.
        // - https://pgbackrest.org/user-guide.html#quickstart/configure-archiving
        // - https://pgbackrest.org/command.html#command-archive-push
        // - https://www.postgresql.org/docs/current/runtime-config-wal.html
        archive := `pgbackrest --stanza=` + DefaultStanzaName + ` archive-push "%p"`
        outParameters.Mandatory.Add("archive_mode", "on")
        outParameters.Mandatory.Add("archive_command", archive)
```
This also requires having a repo retention **and** at least on scheduled backup to get automatic retention management working.

There might be cases where I do not want archiving, eg. loading a large amount of data.
There might be even cases where I do not want archiving nor backups, eg. some kind of temporary test systems.

QUESTION: Is there any way to disable archiving?

### Environment
Please provide the following details:

- Platform: Anthos
- Platform Version: 1.10 (afaik)
- PGO Image Tag: postgres-operator:ubi8-5.1.1-0
- Postgres Version: 14
- Storage: vm ware csi
- Number of Postgres clusters: several

**Comment by laurivosandi** [comment]
I also find it quite odd that I _have to_ enable backups

**Comment by aidensgithub** [comment]
Any solutions found?

**Comment by laurivosandi** [comment]
I mangled the CRD definition and set `minIndex: 0` in the relevant section :D

**Comment by mzwettler2** [comment]
@laurivosandi 

What did you do exactly? Hacking the PGO sources?

**Comment by laurivosandi** [comment]
JFYI: After hacking the CRD-s looks like WAL keeps growing forever :D

So yeah authors should address this issue themselves

**Comment by Pluggi** [comment]
What is the timeline for this feature? Do you have an estimate on when this will be available?

**Comment by sigwinch28** [comment]
Any update on this?

**Comment by bendilley** [comment]
I've found a way to do this temporarily:
```bash
kubectl exec hippo-data-ljc7-0 -c database -it -- bash
vi /pgdata/pg14/postgresql.conf # and set archive_mode = 'off'
pg_ctl restart
```
It's very useful when my initial datasource is a big `pg_dump` file.

After turning archive_mode back on, kicking-off a full backup seems like a good idea:
```bash
kubectl create job --from=cronjob/hippo-repo1-full backup-post-import
```

**Comment by laurivosandi** [comment]
Just move on to https://cloudnative-pg.io/

**Comment by bendilley** [comment]
@mzwettler2 this was with PGO 5.3.0 but I didn't explicitly pause anything.

I do think it's surprisingly difficult to initialise a cluster from a non-PGO datasource. I've even tried restoring from a pgbackrest backup in S3, but because the original source wasn't PGO-managed I still couldn't get it to work as a `dataSource`. `pg_restore` does work, but a large dataset will fill the disk with transaction logs way faster than they can be streamed by pgbackrest, at least to S3.

@laurivosandi thanks for the link - I'm open to alternatives!

**Comment by bendilley** [comment]
Now that I think about it, I did also merge the below patch of the patroni dynamicConfiguration in an earlier experiment - it didn't appear to have any effect, but it _was_ still in place when I manually turned off archive_mode, which might explain why I got away with it 🤷🏻 
```yaml
spec:
  patroni:
    dynamicConfiguration:
      postgresql:
        parameters:
          archive_mode: 'off'
```

**Comment by mzwettler2** [comment]
@bendilley 

Yes, somewhat off topic here. Anyway, I would go with pg_dump/pg_restore. Seems to be the cleanest way for me.

**Comment by bendilley** [comment]
`pg_restore` works fine for smaller datasets, but migrating a large database will fill up the disk with WAL faster than it can be backed-up, at least that's my experience.

**Comment by mzwettler2** [comment]
@bendilley 

Yep. The only "stable" dirty hack I found to prevent for a filesystem full error has been soft-linking a subdirectory of the "/pgbackrest/repo1/archive/db" on the repo pod to "/dev/null". PostgreSQL generates archives and saves them to trash. PGO does not recognize nor undo this filesystem "change".

**Comment by mzwettler2** [comment]
@benjaminjb 

We are hopeing and praying for this feature. When will it come?

**Comment by bendilley** [comment]
Further to my [earlier observation](#issuecomment-1536509116), I've found this combination of steps to be consistently effective:
```bash
kubectl patch postgrescluster hippo --type merge --patch-file patch-patroni-archive-mode.yml
kubectl exec hippo-data-ljc7-0 -c database -it -- bash
vi /pgdata/pg14/postgresql.conf # and set archive_mode = 'off'
pg_ctl restart
exit
```
where `patch-patroni-archive-mode.yml` is 
```yaml
spec:
  patroni:
    dynamicConfiguration:
      postgresql:
        parameters:
          archive_mode: 'off'
```
I've only ever done it while `pg_restore`ing a dump file. Presumably the setting wouldn't persist if the container was restarted.

After the restore is complete, I just reverse the changes and kick-off a full backup:
```bash
kubectl exec hippo-data-ljc7-0 -c database -it -- bash
vi /pgdata/pg14/postgresql.conf # and set archive_mode = 'on'
pg_ctl restart
exit

kubectl patch postgrescluster hippo --type json -p '[{"op": "remove", "path": "/spec/patroni/dynamicConfiguration/postgresql/parameters/archive_mode"}]'
kubectl create job --from=cronjob/hippo-repo1-full backup-post-import
```

**Comment by andrewlecuyer** [comment]
Just dropping a quick update on this thread to note that we are currently taking a look at some upcoming plans and changes to CPK to allow allow additional flexibility around disabling archiving (as requested in this issue).

Therefore, I simply wanted to note that this issue is definitely on our radar, and we will be in touch with additional information, details, etc. once available.

**Comment by dsessler7** [comment]
I'm pleased to announce that support for optional backups has been released with Crunchy Postgres for Kubernetes v5.7.0. Please see the [docs](https://access.crunchydata.com/documentation/postgres-operator/latest/guides/optional-backups) for more info. If you get a chance to try it out, please let us know what you think!

**Comment by mzwettler2** [comment]
@dsessler7 

Thanks for the feature. I will try it out when I find enough time.

I quickly skimmed the documentation. I would have preferred a simple spec.backups.status switch with a three-way logic. 

on: archives + backups (default)
off: no archives + backups
delete: no archives + backups; delete all existing backup artifacts
